view/adminhtml/templates/store_select.phtml

```diff
@@ -89,13 +93,22 @@ $block = $block;
       minimumResultsForSearch: Infinity, // disable search
       width: '25%'
     });
+
     $("#ns8-store-select").on("change", function() {
       var storeId = this.value.toString();
       var store = stores.find(function(store) {
         return store.id === storeId;
       });
-      renderProtectIFrame(store.token);
+      if(storeId !== paramStoreId) {
+        const newUrl = '<?=$block->getUrl('*/*/*')?>' + '?store_id=' + storeId;
+        window.location = newUrl;
+      }else{
+        renderProtectIFrame(store.token);
+      }
     });
+
+    $("#ns8-store-select").val(paramStoreId)
+
     $("#ns8-store-select").trigger("change");
```

Comment:
This shouldn't be necessary anymore with `.val()` call (unless `paramStoreId` is nullable, in which case we probably want to check that before `.val()` call anyways)